components/prism-solidity.js

```diff
@@ -0,0 +1,37 @@
+Prism.languages.solidity = Prism.languages.extend('javascript', {
+	'keyword': [
+		/\b(?:assert|break|constant|continue|do|else|emit|external|for|function|if|import|inherited|internal|is|memory|new|payable|private|public|pure|require|returns?|revert|selfdestruct|storage|suicide|this|throw|var|view|while|_)\b/,
+		{
+			pattern: /\b(?:contract|interface|library|using|struct|function|modifier)\s+[A-Za-z_]\w*(?:\s+is\s+(?:[A-Za-z_][,\s]*)*)?\b/,
+			inside: {
+				'variable': {
+					pattern: /\b(contract|interface|library|using|struct|function|modifier)\s+[A-Za-z_]\w*(?:\s+is\s+(?:[A-Za-z_][,\s]*)*)?\b/,
+					lookbehind: true
+				}
+			}
+		},
+	],
+	'constant': [
+		{
+			pattern: /\b(?:address|string|bytes\d*|int\d*|uint\d*|bool|u?fixed\d+x\d+)\b(?:\s+(?:indexed\s+)?[A-Za-z_]\w*\s*[,)])?/,
+			inside: {
+				'attr-name' : {
+					pattern: /\b(address|string|bytes\d*|int\d*|uint\d*|bool|u?fixed\d+x\d+)\b(?:\s+(?:indexed\s+)?[A-Za-z_]\w*\s*[,)])?/,
```

Comment:
I'm not really sure but isn't `(?:indexed\s+)?` supposed to be part of the lookbehind?

Also, shouldn't `\s*[,)]` be a lookahead rather than being part of the name?

Why is the whole non-lookbehind expression optional? This might cause empty tokens.

Reply:
Actually, this part is from the original patch from the original author, have to admit I didn't pay much attention to it. I think the original author made a mistake. Not sure exactly what besides a simple attribute declaration he intended to match. I'll remove what I think is unnecessary to make it work and update the patch.